src/MongoDB.Driver.Core/Core/NativeLibraryLoader/RelativeLibraryLocatorBase.cs

```diff
@@ -35,31 +36,39 @@ public virtual Assembly GetLibraryBaseAssembly()
 
         public virtual string GetLibraryBasePath()
         {
-            var assembly = GetLibraryBaseAssembly();
-            var codeBase = assembly.CodeBase;
-            var uri = new Uri(codeBase);
-            var absolutePath = uri.AbsolutePath;
-            return Path.GetDirectoryName(absolutePath);
+            // In the nuget package, get the shared library from a relative path of this assembly
+            // Also, when running locally, get the shared library from a relative path of this assembly
+            var assembly = typeof(LibraryLoader).GetTypeInfo().Assembly;
```

Comment:
In the interest of time, using the same approach as the FLE C# binding

Reply:
I used this approach(with location) originally, but then I faced with the issue which is described here: https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/82#discussion_r347569942
not sure yet, how did you fix it

Reply:
We might have to disable shadowcopying to get the tests to run on net452 , good catch: I didn't see this issue since I've configured Rider to disable shadowcopying